app/lib/proxy_api/salt.rb

```diff
@@ -49,7 +49,7 @@ def states_list
       states = {}
 
       environments_list.each do |environment|
```

Comment:
Another way to avoid it would be to expand the `environments` API call to allow providing all the information. That way you don't end up with n+1 API calls but instead just call 1. That probably needs modification on the Smart Proxy side though.

Reply:
IMHO, we can keep that in mind for the future. If the simple fix provided by @evgeni is good enough for now, I would go with that one and bump a new version once it is merged (preferably today - then the packaging bot can pick it up over the weekend).

Reply:
Agreed, it's a nice performance fix but short term I'd merge this bugfix

Reply:
@nadjaheitmann I was assuming you had merge permissions here, but do you?

Reply:
> @nadjaheitmann I was assuming you had merge permissions here, but do you?

@ekohl Yes, I do. Then I would just go ahead and merge this.